### PR TITLE
{chem}[foss/2022b] VAMPIRE-ASM v6.0

### PR DESCRIPTION
**Pull request by guacke**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/v/VAMPIRE-ASM/VAMPIRE-ASM-6.0-foss-2022b.eb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+# License::   MIT/GPL
+
+easyblock = 'ConfigureMake'
+
+name = 'VAMPIRE-ASM'
+version = '6.0'
+
+homepage = 'https://vampire.york.ac.uk/'
+description = """
+Vampire is designed from the ground-up to be an easy to use, fast, open-source and extensible software package capable of modelling almost any magnetic material with atomic resolution.
```
Comment (laraPPr) [inline easybuild/easyconfigs/v/VAMPIRE-ASM/VAMPIRE-ASM-6.0-foss-2022b.eb:11]:
```suggestion
Vampire is designed from the ground-up to be an easy to use, fast, 
open-source and extensible software package capable of modelling 
almost any magnetic material with atomic resolution. 
```

File: easybuild/easyconfigs/v/VAMPIRE-ASM/VAMPIRE-ASM-6.0-foss-2022b.eb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+Vampire is designed from the ground-up to be an easy to use, fast, open-source and extensible software package capable of modelling almost any magnetic material with atomic resolution. 
+"""
+
+toolchain = {'name': 'foss', 'version': '2022b'}
+toolchainopts = {'usempi': True}
+
+source_urls = ['https://github.com/richard-evans/vampire/archive/refs/tags/']
+sources = ['v%(version)s.tar.gz']
+patches = [
+        'VAMPIRE-ASM-6.0_makefile.patch'
```
Comment (laraPPr) [inline easybuild/easyconfigs/v/VAMPIRE-ASM/VAMPIRE-ASM-6.0-foss-2022b.eb:20]:
```suggestion
    'VAMPIRE-ASM-6.0_makefile.patch'
```

File: easybuild/easyconfigs/v/VAMPIRE-ASM/VAMPIRE-ASM-6.0-foss-2022b.eb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+]
+checksums = [
+    {'v6.0.tar.gz': '5a5104eebe6f2c83c11139a6d56cfd78018aba247fd0466e87daa9c39aa62cf8'},
+    {'VAMPIRE-ASM-6.0_makefile.patch':
+    '7c0865ce54993f15ee129a45593b9219c477103ece3f591eff0531f49e337c80'},
```
Comment (laraPPr) [inline easybuild/easyconfigs/v/VAMPIRE-ASM/VAMPIRE-ASM-6.0-foss-2022b.eb:25]:
```suggestion
        '7c0865ce54993f15ee129a45593b9219c477103ece3f591eff0531f49e337c80'},
```

File: easybuild/easyconfigs/v/VAMPIRE-ASM/VAMPIRE-ASM-6.0-foss-2022b.eb [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+skipsteps = ['configure']
+installopts = 'PREFIX=%(installdir)s'
+
+sanity_check_paths = {
+    'files': [
+        'bin/vampire-serial',
+    ],
+    'dirs': [
+        'bin',
```
Comment (laraPPr) [inline easybuild/easyconfigs/v/VAMPIRE-ASM/VAMPIRE-ASM-6.0-foss-2022b.eb:36]:
```suggestion
        'bin',
```

File: easybuild/easyconfigs/v/VAMPIRE-ASM/VAMPIRE-ASM-6.0_makefile.patch [hunk 5]
```diff
@@ -0,0 +1,28 @@
+--- vampire-6.0/makefile.orig	2024-04-16 09:59:14.730461427 +0200
```
Comment (laraPPr) [inline easybuild/easyconfigs/v/VAMPIRE-ASM/VAMPIRE-ASM-6.0_makefile.patch:1]: Please add a despription to the patch. See https://github.com/easybuilders/easybuild-easyconfigs/blob/develop/easybuild/easyconfigs/a/AMOS/AMOS-3.1.0_link_correctly_to_DELTAFILER_SHOWCOORDS.patch for an example.

**Review by laraPPr — CHANGES_REQUESTED** [review]

**Comment by jfgrimm** [comment]
I would suggest switching to the `MakeCp` easyblock -- that way, we don't need to patch the makefile. Instead, something like:
```
files_to_copy = [
    (['vampire-serial', 'vampire-parallel', 'util/vdc/vdc'], 'bin'),
    (['input', 'Co.mat'], 'examples'),
    'license', 'BSD_licence', 'readme.md',
]
```
would do the trick

**Comment by jfgrimm** [comment]
Test report by @jfgrimm
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
gpu25.viking2.yor.alces.network - Linux Rocky Linux 8.9, x86_64, AMD EPYC 7413 24-Core Processor, 1 x NVIDIA NVIDIA H100 PCIe, 535.161.08, Python 3.6.8
See https://gist.github.com/jfgrimm/04faeef7944a4c156cefeb1eeafabfe9 for a full test report.

File: easybuild/easyconfigs/v/VAMPIRE-ASM/VAMPIRE-ASM-6.0_makefile.patch (hunk 5 above)
Comment (guacke) [reply easybuild/easyconfigs/v/VAMPIRE-ASM/VAMPIRE-ASM-6.0_makefile.patch]: Switched to `MakeCp` easyblock (and deleted patch) as suggested by @jfgrimm.

**Comment by jfgrimm** [comment]
@boegelbot: please test @ jsc-zen3

**Comment by jfgrimm** [comment]
@boegelbot: please test @ generoso

**Review by jfgrimm — APPROVED** [review]
lgtm

**Comment by jfgrimm** [comment]
Going in, thanks @guacke!

**Pull request merged**